crates/editor/src/editor.rs

```diff
@@ -7287,17 +7287,58 @@ impl Editor {
         let buffer = self.buffer.read(cx);
         let head = self.selections.newest::<usize>(cx).head();
         let (buffer, head) = buffer.text_anchor_for_position(head, cx)?;
+        let head_offset = text::ToOffset::to_offset(&head, buffer.read(cx));
         let replica_id = self.replica_id(cx);
 
         let workspace = self.workspace()?;
         let project = workspace.read(cx).project().clone();
         let references = project.update(cx, |project, cx| project.references(&buffer, head, cx));
-        Some(cx.spawn(|_, mut cx| async move {
-            let locations = references.await?;
+        Some(cx.spawn(|editor, mut cx| async move {
+            let mut locations = references.await?;
+
+            // Lsp may return references that contain self pos (eg. rust-analyzer)
+            // So we remove self pos from locations
+            editor.update(&mut cx, |_, cx| {
```

Comment:
Does this need to be in an `editor.update`?

Reply:
I've just started working on this project and might not be very familiar with all aspects of it. How should I obtain the `ViewContext` from `AsyncWindowContext` for calling `r.range.to_offset`?

Reply:
Hey, there's a `update` method on AsyncWindowContext that takes a `FnOnce(&mut WindowContext) -> R`.